swfo/h5utils.py

```diff
@@ -34,95 +51,283 @@ def _append_id(h5_key):
     return max(ids) + 1
 
 
-def write_h5_md(
+def _write_dataset(
         h5_group: h5py.Group,
-        datasets: Union[List[Dict], Dict],
-        dataset_names: Optional[List[str]] = None) -> None:
+        dataset: Dict,
+        dataset_path: str = '',
+        track_order: bool = True) -> h5py.Dataset:
+    """
+    Internal function for writing dataset metadata to a H5file
+
+    :param h5_group:
+        Root reference for the h5 collection
+    :param dataset:
+        metadata encoded as a nested dictionary
+    :param dataset_path:
+        dataset path corresponding to the metadata
+    :param track_order:
+        flag to track insertion order on h5Groups
     """
-    Serialises provided metadata information to yaml format inside hdf5
-    archive.
-    If a single dataset is provided the metadata is written to /METADATA/_{index}
-    and linked to /METADATA/CURRENT
-
-    If multiple datasets are provided and named, they are written to
-        /METADATA/{dataset_name}/_{index}
-    linked to /METADATA/{dataset_name}/CURRENT and
-    consolidated in a virtual dataset under /METADATA/CURRENT
+    if dataset_path[0] == '/':  # os.path.join can't be provided absolute refs
```

Comment:
@uchchwhash had a better solution here for calling lstrip or something to handle absolute paths; not sure if we want to propagate it here

Reply:
Also unsure if we should merge these changes in through the separate PR